src/adapters/mc/OsConfigResource.c

```diff
@@ -328,12 +366,132 @@ static MI_Result SetDesiredObjectValueToDevice(const char* who, char* objectName
         payloadSize = (int)strlen(serializedValue);
         if (NULL != (payloadString = malloc(payloadSize + 1)))
         {
+            const OsConfigComponent *component = NULL;
+
             memset(payloadString, 0, payloadSize + 1);
             memcpy(payloadString, serializedValue, payloadSize);
 
-            mpiResult = AsbMmiSet(g_componentName, objectName, payloadString, payloadSize, GetLog());
-            LogInfo(context, GetLog(), "[%s] AsbMmiSet(%s, %s, '%.*s', %d) returned %d",
-                who, g_componentName, objectName, payloadSize, payloadString, payloadSize, mpiResult);
+            for (component = g_components; NULL != component->Name; ++component)
+            {
+                if (!strcmp(component->Name, g_componentName))
+                {
+                    break;
+                }
+            }
+            if (NULL != component->MmiSet)
+            {
+                mpiResult = component->MmiSet(g_componentName, objectName, payloadString, payloadSize, GetLog());
```

Comment:
>component->MmiSet( [](http://example.com/codeflow?start=28&length=18)

Adapters MUST NOT call MMI directly. Adapters can only call the MPI. In this case, the Universal NRP adapter can call OSConfig over the MPI REST API.